multimodal/src/autogluon/multimodal/matcher.py

```diff
@@ -1153,36 +1224,70 @@ def _predict(
 
         return outputs
 
+    def _evaluate_symmetric_ranking(self, data):
+        data_with_label, query_data, response_data, label_column = convert_data_for_ranking(
+            data=data,
+            query_column=self._query[0],
+            response_column=self._response[0],
+        )
+        logger.debug(f"first _evaluate_ranking...\n")
+        score_1 = self._evaluate_ranking(
+            qr_relevance=data_with_label,
+            query_data=query_data,
+            response_data=response_data,
+            label_column=label_column,
+            cutoff=[1, 5, 10],
+        )
+        data_with_label, query_data, response_data, label_column = convert_data_for_ranking(
+            data=data,
+            query_column=self._response[0],
+            response_column=self._query[0],
+        )
+        logger.debug(f"second _evaluate_ranking...\n")
+        score_2 = self._evaluate_ranking(
+            qr_relevance=data_with_label,
+            query_data=query_data,
+            response_data=response_data,
+            label_column=label_column,
+            cutoff=[1, 5, 10],
+        )
+
+        return sum(score_1.values()) + sum(score_2.values())
+
     def _evaluate_ranking(
         self,
         qr_relevance: Union[pd.DataFrame, dict, list],
         query_data: Union[pd.DataFrame, dict, list],
         response_data: Union[pd.DataFrame, dict, list],
+        label_column: str,
```

Comment:
is label_column compulsory?

Reply:
I think so. We need to know the label column in `qr_relevance`. For datasets without `label_column` in training, we still need it in evaluation.